test/unit/io/sam_file/format_sam_test.cpp

```diff
@@ -176,13 +260,72 @@ TEST_F(sam_format, header_errors)
     {
         std::string header_str
         {
-            "@HD\tVN:1.0\n"
-            "@SQ\tSN:ref\tLN:0\n" /*wrong length*/
+            "@SQ\tSN:ref\tLN:0\n"                           // allowed range of LN: [1,2^31−1]
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "The required LN tag in @SQ is missing or 0."
+    }
+    {
+        std::string header_str
+        {
+            "@SQ\tSN:ref\tLN:-1\n"                          // allowed range of LN: [1,2^31−1]
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "Provided reference has unequal length as specified in
+                                                            //  the header."
+    }
+    {
+        std::string header_str
+        {
+            "@SQ\tSN:ref\tLN:2147483647\n"                  // = 2^31−1, allowed range of LN: [1,2^31−1]
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_NO_THROW(fin.begin());
+    }
+    {
+        std::string header_str
+        {
+            "@SQ\tSN:ref\tLN:2147483648\n"                  // = 2^31, allowed range of LN: [1,2^31−1]
+        };
+        std::istringstream istream(header_str);
+        seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "[CORRUPTED SAM FILE] Casting '2147483648' into type int
+                                                            //  would cause an overflow."
+    }
+    {
+        std::string header_str
+        {
+            "@SQ\tSN:ref\tLN:4\n"
         };
         std::istringstream istream(header_str);
         seqan3::sam_file_input fin{istream, this->ref_ids, this->ref_sequences, seqan3::format_sam{}};
-        EXPECT_THROW(fin.begin(), seqan3::format_error);
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "Provided reference has unequal length as specified in
+                                                            //  the header."
```

Comment:
```suggestion
        EXPECT_THROW(fin.begin(), seqan3::format_error);
```

(the message is also different now)